report/CommunicationStrategies/usecases.tex

```diff
@@ -0,0 +1,38 @@
+
+\subsection{Actors}
+In more general terms a actor is something that can interact with the system. That could be a person, a machine or another software running on the same machine. 
+A key destingtion to make is that a, ex. person, is not limited to be only one type of actor.
+Actors should more be regarded as roles in the system, so if Britta works at a store as a clerk, she will have the actor type clerk when she is working, but in her freetime she might shop in the same store and thereby take the actor role as customer. 
+
+In a broader sense we can consider actors to be everything communicating with the system from outside the system it self.
+
+In our solution we can use the notion of actors to identify roles with in the SUD, this will allow the developers and customers to communicate easier since they will have a unified vocabulary when talking about who communicates with the system. 
+It will also help if the SUD needs boundaries with in the system, i.e. functionalities only avaible to some actors, the customers will then be able to specify a list of actors who should have access to said functionalities, which will help the developers in ensuring correctness.
+The developers might even create a access layer with in the SUD based of the actors identified in the requirements or groupings of actors.
+
+\subsection{Usecases}
+A usecase is a complete set of events that specifies all the action needed for a perticular actor to achieve a goal with in the system. 
+The set of events should be considered as transaktions - meaning, first the actor does something, then the system respons, then the actor does something, then the system respons and so forth. 
+A usecase is always from the perspective of a perticular actor, therefore it is important to only focus on details relevant for said actor. 
+A actor will have many usecases and a there will be many usecases similiar to one or another - that could be multiple actors having similar usecases or many usecases having similar event traces. 
+The act of defining usecases can help discover requirements for the system, since it forces one to think about how each actor is suppose to interact with the system.
+
+Since the usecases are a complete trace of events for a perticular action or goal, a single usecase can then be considered as a transaction system describing a feature og functionality of the system. 
+The group of all usecases then specifies all the features and functionalities of the system and how they should work. 
+This allows us to consider our software as a set of state transition graphs where stimulus between the actors and the system determine the behaviour of the system.
+
+The notion of usecases does not fit perfectly into our project, since we suspect that it would be impossible for a customer to know how to construct the event traces in a realistic manner. 
```

Comment:
```suggestion
The notion of use cases does not fit perfectly into our project, since we suspect that it would be impossible for a customer to know how to construct the event traces in a realistic manner. 
```